src/iexamine.cpp

```diff
@@ -3497,6 +3498,80 @@ void iexamine::climb_down( player &p, const tripoint &examp )
     g->m.creature_on_trap( p );
 }
 
+void iexamine::autodoc( player &p, const tripoint &examp ) {
+    enum options {
+        CONFIGURE,
+        INSTALL_CBM,
+        CANCEL,
+    };
+
+    std::string status = uistate.iexamine_autodoc_config ? _( "Online" ) :
+                         _( "Configuration required" );
+
+    uimenu amenu;
+    amenu.selected = 0;
+    amenu.text = string_format( _( "Autodoc Mk. XI.  Status: %s.  Please choose operation." ), status );
+    amenu.addentry( CONFIGURE, true, 'c', str_to_illiterate_str( _( "Configure Autodoc." ) ) );
+    amenu.addentry( INSTALL_CBM, uistate.iexamine_autodoc_config, 'i',
+                    str_to_illiterate_str( _( "Choose Compact Bionic Module to install." ) ) );
+
+    amenu.addentry( CANCEL, true, 'q', str_to_illiterate_str( _( "Do nothing." ) ) );
+
+    amenu.query();
+
+    bool adjacent_couch = false;
+    bool in_position = false;
+    for( const auto &couch_loc : g->m.points_in_radius( examp, 1, 0 ) ) {
+        if( g->m.furn( couch_loc ).obj().id == "f_autodoc_couch" ) {
```

Comment:
You don't need to get the `furn_t` object to get its id. `map::furn` returns an id that you can compare directly. I suggest you store "f_autodoc_couch" as local constant of type `furn_str_id` and compare it to the result of `map::furn`.